node/actors/bft/src/chonky_bft/proposer.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{metrics, Config, OutputSender};
+use std::sync::Arc;
+use zksync_concurrency::{ctx, error::Wrap as _, sync, time};
+use zksync_consensus_network::io::ConsensusInputMessage;
+use zksync_consensus_roles::validator;
+
+/// Timeout for creating a proposal. If the proposal is not created in this time, the proposer
+/// will quit trying to create a proposal for this view. This can be different from the replica
+/// timeout for the whole view.
+pub(crate) const PROPOSAL_CREATION_TIMEOUT: time::Duration = time::Duration::milliseconds(2000);
```

Comment:
why do we need this? Wouldn't just the view timeout work?

Reply:
It can. I thought it might be useful in the future for sticky leaders. If we want to pipeline block production.

Reply:
I'd remove it for now then, but up to you